cli/kubernetes/default_options_reader.go

```diff
@@ -0,0 +1,27 @@
+package kubernetes
+
+type DefaultOptionsReader struct{}
+
+// NewDefaultOptionsReader is a reader used by the Installer to
+// fill InstallationOptions with an initial default value, if such
+// is specified (dynamic per function, or static).
+func NewDefaultOptionsReader() DefaultOptionsReader {
+	return DefaultOptionsReader{}
+}
+
+// Read attempts to fill the option with a default, dynamic or static.
+func (reader DefaultOptionsReader) Read(option *InstallationOption) error {
+	// Give priority to a function which provides the default
+	// value dynamically.
+	if option.DynDefault != nil {
+		err := option.DynDefault(option)
```

Comment:
This allows one to call a `DynDefault` function like this: 

```
option1.DynDefault(option2)
```
which of course makes little sense. We can make the invocation more intuitive by creating a "method" on the InstallationOption that calls the function set as an attribute:

```
func (option *InstallationOption) DynDefault() {
  option.DynDefaultFunc(option)
}
```
then [here](https://github.com/SUSE/carrier/pull/30/files#diff-c762dd02a47c7bb43d19bcf19914b1cc6294308fbcc4e1c2cfd4ecf4ddc3290aR71) you set the `option.DynDefaultFunc` and the invocation would then be:

```
  option.DynDefault()
```

It's more code for the same thing but I think it improves the readability of the invocation by wrapping the "ugly" part in a pre-defined function.

I like the function attribute idea for the default, +1.

Reply:
Agreed. Missed that my code allows for option/argument mismatch.

Reply:
Done with commit 40a6401915c80f94e70b7575be98ab7224829041